components/dashboard/src/components/Modal.tsx

```diff
@@ -150,7 +150,7 @@ export const ModalFooter: FC<ModalFooterProps> = ({ error, warning, children })
     return (
         <div className="relative">
             {hasAlert && (
-                <div className="absolute -top-12 left-0 right-0" style={alertStyles}>
+                <div className="absolute bottom-12 left-0 right-0" style={alertStyles}>
```

Comment:
Positioning off of the bottom allows it to grow taller if needed (still some styling issues for long error messages, but this should be a small improvement).

Reply:
**issue**: Using the alert component here sounds perfect, however, using alerts or error feedback inside modals can be tricky, especially after adding a scrollable area inside the modal in https://github.com/gitpod-io/gitpod/pulls?page=3&q=is%3Apr+author%3Aselfcontained+is%3Aclosed.

Reply:
Yup, I think having the alert positioned absolutely somewhere so it's kind of pinned at the footer is one way to avoid some of those issues (but does introduce some other challenges). Currently I'm using a pattern where the alert is pinned right above the footer (part of the footer actually). I've added some extra padding to the bottom of the modal body section to account for it, but that's a temporary solution and not ideal. What I'd like to do:

* Animate the alert so it slides up from the footer.
* Find a more robust way to deal with the alert potentially blocking content in the modal body. Some ideas:
  * Dynamically adding padding to the bottom of the modal body based on the size of the alert when it's present so content can be scrolled to in the body.
  * Making the pinned modal alert dismissible (probably the easiest solution to implement).